test/integration/Main.hs

```diff
@@ -248,7 +248,7 @@ testScenarioSolution _ci _em =
       Left x -> assertFailure $ unwords ["Failure in initGameStateForScenario:", T.unpack x]
       Right gs -> case gs ^. winSolution of
         Nothing -> assertFailure "No solution to test!"
-        Just sol@(ProcessedTerm _ _ _ reqCtx) -> do
+        Just sol@(ProcessedTerm _ _ reqCtx) -> do
```

Comment:
Not a single new test is needed? :open_mouth: At least a small unit test to showcase the usage would be nice.

Reply:
Oh, very good point!  I was so focused on the refactoring, and so elated when I finally got everything compiling and passing all the tests again, that I totally forgot about adding some tests. :sweat_smile:  I will add some.

Reply:
Added some basic tests.